oauth2_http/java/com/google/auth/oauth2/UserCredentials.java

```diff
@@ -89,13 +102,21 @@ private UserCredentials(
       String clientSecret,
       String refreshToken,
       AccessToken accessToken,
+      Collection<String> scopes,
```

Comment:
If this collection is passed in from outside, this could be a security issue

Reply:
Exact same behavior as ServiceAccountCredentials class. 

ServiceAccountCredentials have protected constructor. In this class the constructor is private. Can be only called by the Builder.